gson/src/main/java/com/google/gson/internal/reflect/UnsafeReflectionAccessor.java

```diff
@@ -79,7 +79,7 @@ private static Object getUnsafeInstance() {
   private static Field getOverrideField() {
     try {
       return AccessibleObject.class.getDeclaredField("override");
-    } catch (NoSuchFieldException e) {
+    } catch (Exception e) {
```

Comment:
I think it would be better to explicitly catch `NoSuchFieldException` and `SecurityException` to make the intention more clear (though the other methods in that class simply catch `Exception` as well).
Sadly multi-catch was added in Java 7, but since the code in the `catch` block is only a simple return, I think it would be okay to duplicate it.

Reply:
hmm.... personally, I'd prefer to leave as is. this access being blocked by my security policy was the specific reason for me to make this change, but I want to try to use the fallback behavior in `makeAccessible` if we should fail to retrieve this field for _any_ reason whatsoever. this increases Gson's chances of continuing to work in strange environments or future JDKs and makes the failure reason and resolution steps more obvious when it can't (compare the nice, readable exception thrown in `makeAccessible` to the arcane stack trace barfed up by ReflectionAccessor's class initializer)

I'm less concerned with signaling intent (which imo is still obvious) than I am with preserving function.